radicle-node/src/service.rs

```diff
@@ -905,12 +907,22 @@ where
 
     /// Announce local refs for given id.
     fn announce_refs(&mut self, id: Id) -> Result<(), storage::Error> {
+        type Refs = BoundedVec<Id, REF_LIMIT>;
+
         let node = self.node_id();
         let repo = self.storage.repository(id)?;
         let remote = repo.remote(&node)?;
         let peers = self.sessions.negotiated().map(|(_, _, p)| p);
-        let refs = remote.refs.into();
         let timestamp = self.clock.timestamp();
+
+        if remote.refs.len() > Refs::max() {
```

Comment:
Why do we not want to send multiple announcements instead? Dropping refs seems strange to me.

Reply:
True, for InventoryAnnouncement we couldn't do that because it's supposed to be the entire inventory, but for refs I think we could? Would have to think through this a bit more..